modules/shapefile/src/lib/parsers/parse-shp.js

```diff
@@ -111,7 +127,7 @@ function parseState(state, result = {}, binaryReader) {
               type: recordHeaderView.getInt32(8, LITTLE_ENDIAN)
             };
 
-            if (!binaryReader.hasAvailableBytes(recordHeader.byteLength - SHP_RECORD_HEADER_SIZE)) {
+            if (!binaryReader.hasAvailableBytes(recordHeader.byteLength - 4)) {
```

Comment:
I was previously checking for too few bytes to exist. `binaryReader.getDataView(SHP_HEADER_SIZE);` is already called before this line, so I don't need to subtract it.

I'm able to subtract `4` here because `binaryReader.rewind(4);` is called below before `binaryReader.getDataView(recordHeader.byteLength);`